src/VisualStudio/Core/Def/Implementation/Options/LocalUserRegistryOptionPersister.cs

```diff
@@ -77,24 +77,41 @@ bool IOptionPersister.TryFetch(OptionKey optionKey, out object value)
                     }
                     else if (optionKey.Option.Type == typeof(long))
                     {
-                        object untypedValue = subKey.GetValue(key, defaultValue: optionKey.Option.DefaultValue);
-
-                        if (untypedValue is string stringValue)
+                        var untypedValue = subKey.GetValue(key, defaultValue: optionKey.Option.DefaultValue);
+                        switch (untypedValue)
                         {
-                            // Due to a previous bug we were accidentally serializing longs as strings. Gracefully convert
-                            // those back.
-                            bool suceeded = long.TryParse(stringValue, out long longValue);
-                            value = longValue;
-                            return suceeded;
+                            case string stringValue:
+                                {
+                                    // Due to a previous bug we were accidentally serializing longs as strings.
+                                    // Gracefully convert those back.
+                                    var suceeded = long.TryParse(stringValue, out long longValue);
+                                    value = longValue;
+                                    return suceeded;
+                                }
+
+                            case long longValue:
+                                value = longValue;
+                                return true;
                         }
-                        else if (untypedValue is long longValue)
+                    }
+                    else if (optionKey.Option.Type == typeof(int))
+                    {
+                        var untypedValue = subKey.GetValue(key, defaultValue: optionKey.Option.DefaultValue);
```

Comment:
I think we want to be super crisp about how things convert to/from registry form.  And, in the case where we've stored data in the wrong form, we want to be crisp about the code that gracefully handles it.